### DISCUSSION TITLE
Shouldn't open source projects also use open source projects for project organisation?

### DISCUSSION THREAD

**Discussion by linux-lukas** [mitmproxy/mitmproxy#5580] [General]
The mitmproxy project site is hosted on the infrastructure of the US BigTech company Amazon.

For communication with developers, the proprietary messaging software application (see https://en.wikipedia.org/wiki/Slack_(software)) is used. 

For public relations, data-guzzling social media platforms are used.

In all of this, I often ask myself whether free and open-source software solutions should not operate more independently of BigTech corporations. 

Example: The project website, email infrastructure, ticket system if necessary, knowledge management, source code provision is not handled on the infrastructure of big tech companies such as Microsoft, Apple, Amazon, Google, etc. Instead, an independent hoster is chosen for the project website, and for communication with developers, but also with the community, a self-hosted Mattermost, Matrix or XMPP instance with corresponding organisational conditions (space, rooms, etc.) would be used instead of Slack. For the public work of the project, free and decentrally organised social networks would be used (keyword: Fediverse).

How do you see it?

**Comment by mhils** [comment]
How I see it:

 - mitmproxy never phones home. There is no telemetry at all.[^1]
 - [mitmproxy.org](https://mitmproxy.org/) does not embed any analytics, trackers, cookie banners, ads, etc. I've recently spent quite some effort to replace the Twitter widget on the front page with a privacy-preserving screenshot (this was our last embedded third party).

From that perspective I think we are doing a reasonable job at protecting user's privacy.
I know very few commercial or FOSS projects that are at the same level.


-------------------------------

> The mitmproxy project site is hosted on the infrastructure of the US BigTech company Amazon.

There are a few reasons for this. First, we want a CDN to handle traffic peaks (Hacker News, new version releases, ...). So if you don't want to just fall over in these cases, there's no way around a big tech company. Second, from a security perspective, using AWS S3 + CloudFront + MFA greatly reduces our attack surface compared to a self-hosted server. We really don't want to ship any compromised binaries.

> For communication with developers, the proprietary messaging software application (see https://en.wikipedia.org/wiki/Slack_(software)) is used.

We migrated off IRC to Slack in 2015 because IRC did not work well for us (e.g. lack of message retention). At the time there were no alternatives that met our needs. While I would probably look towards Zulip (Cloud) today, Slack has really been good enough for us. Our main use case here is 1-on-1 coordination when working on the same parts of the code, everything else on GitHub really. You don't miss anything if you're not on Slack. I'm sure that all main developers would be happy to talk to you on Signal/IRC/GitHub/... instead if you want to contribute code.

> For public relations, data-guzzling social media platforms are used.

Our main communication platform is our blog (https://mitmproxy.org/posts/), which has an RSS feed you can subscribe to. Most of our social media activity is centered around promoting our blog content. I don't think we'd be able to reach a similar audience on the Fediverse.

[^1]: This includes not reporting crashes or doing update checks, which would be very nice to have.

**Reply by linux-lukas** [reply]
Thank you for your feedback. 

You are right that the website is clean in terms of third party integration - thank you for your and your colleagues' work on the website!

> *“There are a few reasons for this. First, we want a CDN to handle traffic peaks (Hacker News, new version releases, ...). So if you don't want to just fall over in these cases, there's no way around a big tech company. Second, from a security perspective, using AWS S3 + CloudFront + MFA greatly reduces our attack surface compared to a self-hosted server. We really don't want to ship any compromised binaries.“*

These are understandable reasons and in this respect there will probably be no alternative in the short and medium term. Unfortunately. But perhaps there will be in the long term. 

> *“We migrated off IRC to Slack in 2015 because IRC did not work well for us (e.g. lack of message retention). At the time there were no alternatives that met our needs. While I would probably look towards Zulip (Cloud) today, Slack has really been good enough for us. Our main use case here is 1-on-1 coordination when working on the same parts of the code, everything else on GitHub really. You don't miss anything if you're not on Slack. I'm sure that all main developers would be happy to talk to you on Signal/IRC/GitHub/... instead if you want to contribute code.“* 

Thank you for this insight! And yet the question of whether Matrix or XMPP might be useful in the long term. It would then have to be seen whether it meets the demands of the project. 

> *“Our main communication platform is our blog (https://mitmproxy.org/posts/), which has an RSS feed you can subscribe to. Most of our social media activity is centered around promoting our blog content. I don't think we'd be able to reach a similar audience on the Fediverse.“*

Well, but at the same time, maybe people from the IT sector and not the IT sector (but e.g. from the public administration sector) could become aware of this project, because he gets a toot of your Mastodon profile flushed into his timeline (TL). 

The RSS feed could be bridged to mastodon.

Finally, I would like to thank you again for your detailed feedback!
